### PR TITLE
Get rid of deprecation warnings for  rawurlencode() and strpos() (#380)

### PR DESCRIPTION
**Pull request by g41797**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #380 

"Deprecated" warning now showed for trim(), but it should be fixed in [UrlManager.php](https://github.com/yiisoft/yii2/blob/master/framework/web/UrlManager.php) 
 (another repo)

devenv:
- PhpStorm 2023.3.1
- PHP 8.3.1 
- Xdebug v3.3.1

### CODE REVIEW

File: src/OAuth1.php [hunk 1]
```diff
@@ -335,17 +335,28 @@ public function signRequest($request, $token = null)
...
         if (strpos($url, '?') !== false) {
             list($url, $queryString) = explode('?', $url, 2);
             parse_str($queryString, $urlParams);
             $params = array_merge($urlParams, $params);
         }
         unset($params['oauth_signature']);
         uksort($params, 'strcmp'); // Parameters are sorted by name, using lexicographical byte value ordering. Ref: Spec: 9.1.1
+
+        $parts0 = strtoupper($method);
+        $parts0 = isset($parts0) ? $parts0 : '';
```
Comment (samdark) [inline src/OAuth1.php:349]: Won't it always be set cause it's defined above?

File: src/OAuth1.php [hunk 2]
```diff
@@ -335,17 +335,28 @@ public function signRequest($request, $token = null)
...
+
+        $parts1 = isset($url) ? $url : '';
+
+        $parts2 = http_build_query($params, '', '&', PHP_QUERY_RFC3986);
+        $parts2 = isset($parts2) ? $parts2 : '';
```
Comment (samdark) [inline src/OAuth1.php:354]: Won't it always be set cause it's defined above?

File: src/OAuth1.php (hunk 1 above)
Comment (g41797) [reply src/OAuth1.php:349]: fixed

File: src/OAuth1.php (hunk 2 above)
Comment (g41797) [reply src/OAuth1.php:354]: fixed

Comment (samdark) [reply src/OAuth1.php:354]: Fixed where?

Comment (g41797) [reply src/OAuth1.php]: [1](https://github.com/yiisoft/yii2-authclient/pull/383/commits/45cd5305eba0c9f684d529d8f0865c24252bfb72)

[2](https://github.com/yiisoft/yii2-authclient/pull/383/commits/46ccb8617a76595fb651a5510a66eca3c91f724a)

File: src/OAuth1.php [hunk 3]
```diff
@@ -335,13 +335,18 @@ public function signRequest($request, $token = null)
      */
     protected function composeSignatureBaseString($method, $url, array $params)
     {
+        $url    = isset($url) ? $url : '';
```
Comment (samdark) [inline src/OAuth1.php:338]:
```suggestion
        $url = isset($url) ? $url : '';
```

**Comment by samdark** [comment]
@g41797 do you want to fix the rest of compatibility issues? See tests failing.

**Comment by samdark** [comment]
Merged that one.

**Pull request closed without merging**